Sources/SwiftKuery/ConnectionPool.swift

```diff
@@ -108,9 +108,23 @@ public class ConnectionPool {
     // needed, so MUST later be returned to the pool with give() if it is to be reused.
     // Items can therefore be borrowed or permanently removed with this method.
     //
-    // This function will block until an item can be obtained from the pool. 
+    // This function will block until an item can be obtained from the pool.
     private func take() -> Connection? {
+        defer {
+            unlockPoolLock()
+        }
```

Comment:
This is too early - you should only `defer { unlockPoolLock() }` _after_ you have done a `lockPoolLock()`.  Otherwise you may unlock when you weren't locked.